go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go

```diff
@@ -669,6 +669,47 @@ func TestSchemaChange(t *testing.T) {
 			})
 		})
 	}
+	// 'mysql' strategy
+	t.Run("mysql strategy", func(t *testing.T) {
+		t.Run("declarative", func(t *testing.T) {
+			t1uuid = testOnlineDDLStatement(t, createT1Statement, "mysql --declarative", "vtgate", "just-created", "", false)
+
+			status := onlineddl.WaitForMigrationStatus(t, &vtParams, shards, t1uuid, normalWaitTime, schema.OnlineDDLStatusComplete, schema.OnlineDDLStatusFailed)
```

Comment:
Curious why we specify multiple statuses in `WaitForMigrationStatus` but then only a single one in `CheckMigrationStatus`?

Reply:
The reason is that we wait till either status is seen (the function will otherwise give some `30sec` for the migration to "evolve"). So there are terminal states for the function, so that it does not need to wait after meeting these states. This just reduces test time. Then, we really want to validate that we reached a particular result: typically we look for a success, ie `OnlineDDLStatusComplete`. But sometimes we intentionally seek a failure.